### PR TITLE
Port cmdLineTester_GCCheck into openj9

### PR DESCRIPTION
**Pull request by OscarQQ**
related: backlog issue 434
Signed-off-by: OscarQQ <Yixin.Qian@ibm.com>

### CODE REVIEW

**Comment by OscarQQ** [comment]
| Build       | Platform           | Java Version  | JDK Impl |
|:-------------:|:-------------:|:-----:|:-----------:|
|https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5967/|ppc64_aix,ppc64le_linux,s390x_linux,x86-64_linux,x86-64_mac,x86-64_windows|8,11|OpenJ9|

File: test/functional/cmdLineTests/gcCheck/build.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		</copy>
+	</target>
+
+	<target name="clean" depends="dist" description="clean up">
+		<!-- Delete the ${build} directory trees -->
+		<delete dir="${build}" />
+	</target>
+
+	<target name="build" >
+		<antcall target="clean" inheritall="true" />
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/build.xml:72]: This is a specific test for openj9/ibm. Please add the if block.

File: test/functional/cmdLineTests/gcCheck/build.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		<echo>============COMPILER SETTINGS============</echo>
+		<echo>===fork:                         yes</echo>
+		<echo>===executable:                   ${compiler.javac}</echo>
+		<echo>===debug:                        on</echo>
+		<echo>===destdir:                      ${DEST}</echo>
+		<javac destdir="${build}" debug="true" fork="true" executable="${compiler.javac}" includeAntRuntime="false" encoding="ISO-8859-1">
+			<src path="${src}" />
+			<src path="${allocator}" />
+			<exclude name="**/TestGCPolicyNogc.java" />
+			<exclude name="**/TestNogcDieWithOOM.java" />
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/build.xml:52]: Is it possible to use include?

File: test/functional/cmdLineTests/gcCheck/playlist.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	${TEST_STATUS}</command>
+		<levels>
+			<level>sanity</level>
+		</levels>
+		<groups>
+			<group>functional</group>
+		</groups>
+		<subsets>
+			<subset>8</subset>
+		</subsets>
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/playlist.xml:51]: This is a specific test for openj9/ibm. Please add impl

File: test/functional/cmdLineTests/gcCheck/playlist.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	${TEST_STATUS}</command>
+		<levels>
+			<level>sanity</level>
+		</levels>
+		<groups>
+			<group>functional</group>
+		</groups>
+		<subsets>
+			<subset>9+</subset>
+		</subsets>
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/playlist.xml:78]: This is a specific test for openj9/ibm. Please add impl

**Comment by llxia** [comment]
Please make sure this test pass on zos and vmfarm. Thanks

**Comment by OscarQQ** [comment]
I have seen intermittent failures on the macos platform e.x. https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5994/, https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/5978/
```
01:27:48  Test result: FAILED
01:27:48  Output from test:
01:27:48   [OUT] success!
01:27:48   [ERR] JVMDUMP039I Processing dump event "vmstop", detail "#0000000000000000" at 2021/01/19 09:27:42 - please wait.
01:27:48   [ERR] JVMDUMP032I JVM requested System dump using '/Users/jenkins/workspace/Grinder/openjdk-tests/TKG/test_output_16110484601484/cmdLineTester_GCCheck_SE80_1/j9core.dmp' in response to an event
01:27:48   [ERR] JVMDUMP012E Error in System dump: The core file created by child process with pid = 24390 was not found. Expected to find core file with name "/cores/core.24390"
01:27:48   [ERR] JVMDUMP013I Processed dump event "vmstop", detail "#0000000000000000".
01:27:48  >> Success condition was not found: [Output match: System dump written]
```
@pshipton could you take a look?

**Comment by pshipton** [comment]
I don't think creating core files on Mac works when you have a signed build (as produced at AdoptOpenJDK) and the OS is 10.15+.  Although the machine seems to be 10.14 in this case. Not sure how Adopt is handling this problem in other testing.

This should work in the builds at OpenJ9 as these builds are not signed. These all passed. 1440 and 1442 are 10.14 machines.
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1436
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1438
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1439
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1440
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1441
https://ci.eclipse.org/openj9/view/Test/job/Grinder/1442

File: test/functional/cmdLineTests/gcCheck/src/org/openj9/test/nogc/Main.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+		allocator.start();
+		System.gc();
+		
+		while (allocator.isAlive()) {
+			try {
+				count++;
+				if (count > 50) {
+					Allocator.bQuit = true;
+				}
+				System.out.println("success!");
```
Comment (pshipton) [inline test/functional/cmdLineTests/gcCheck/src/org/openj9/test/nogc/Main.java:38]: Does this need to print `"success!"` 50 times?

File: test/functional/cmdLineTests/gcCheck/playlist.xml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	<test>
+		<testCaseName>cmdLineTester_GCCheck</testCaseName>
+		<variations>
+			<variation>Mode110</variation>
+			<variation>Mode610</variation>
+			<variation>Mode301</variation>
+			<variation>Mode501</variation>
+			<variation>Mode551</variation>
+		</variations>
+		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -DOPTION_ADDEXPORTS=$(SQ)--add-exports=java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED$(SQ) \
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/playlist.xml:66]: Could you verify if we still need this? If no, then we can merge two test targets.

File: test/functional/cmdLineTests/gcCheck/playlist.xml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	-DEXE=$(SQ)$(JAVA_COMMAND) $(JVM_OPTIONS)$(SQ) \
+    -DJDMPVIEW_EXE=$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jdmpview$(EXECUTABLE_SUFFIX)$(Q) \
+	-DRESJAR=$(Q)$(CMDLINETESTER_RESJAR)$(P)$(TEST_RESROOT)$(D)gcCheck.jar$(Q) \
+	-Xint -jar $(CMDLINETESTER_JAR) \
+	-config $(Q)$(TEST_RESROOT)$(D)gcchecktests.xml$(Q) -verbose -explainExcludes \
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/playlist.xml:71]: Please remove  -verbose

File: test/functional/cmdLineTests/gcCheck/playlist.xml [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+		<variations>
+			<variation>Mode110</variation>
+			<variation>Mode610</variation>
+			<variation>Mode301</variation>
+			<variation>Mode501</variation>
+			<variation>Mode551</variation>
+		</variations>
+		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -DOPTION_ADDEXPORTS=$(SQ)--add-exports=java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED$(SQ) \
+	-DEXE=$(SQ)$(JAVA_COMMAND) $(JVM_OPTIONS)$(SQ) \
+    -DJDMPVIEW_EXE=$(Q)$(TEST_JDK_HOME)$(D)bin$(D)jdmpview$(EXECUTABLE_SUFFIX)$(Q) \
```
Comment (llxia) [inline test/functional/cmdLineTests/gcCheck/playlist.xml:68]: Please fix indentation

**Comment by fjeremic** [comment]
The PR title is a bit misleading. I thought at first we're talking about the `!gccheck` `jdmpview` command, but in fact this porting the GCCheck tests into OpenJ9. Could we adjust the PR title, and while at it fix up the commit title as well to be consistent with the contribution guidelines:

> The first line should be less than 70 characters, where reasonable, and should be written in sentence case **(capitalize the first letter)** preferably not ending in a period.

https://github.com/eclipse/openj9/blob/master/CONTRIBUTING.md#commit-guidelines

**Review by fjeremic — APPROVED** [review]

**Comment by llxia** [comment]
Jenkins test sanity win,xlinux,aix jdk8,jdk11

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**